src/Cmp/Queues/Infrastructure/AWS/v20121105/Queue/MessageHandler.php

```diff
@@ -36,9 +39,12 @@ public function handleMessage(array $message)
             throw new ReaderException("Handling a message with no callback set");
         }
 
-        $body = json_decode($message['Body'], true);
-        $task = $this->jsonMessageFactory->create($body['Message']);
-        call_user_func($this->callback, $task);
+        try{
+            $body = json_decode($message['Body'], true);
+            call_user_func($this->callback, $this->jsonMessageFactory->create($body['Message']));
+        } catch(Exception $e){
```

Comment:
This catch is assuming too many things! Not an exception will be a ParseMessageException always.